src/compiler/crystal/codegen/debug.cr

```diff
@@ -76,11 +76,10 @@ module Crystal
     end
 
     def current_debug_file
-      if location = @current_debug_location
-        debug_files_cache[location.filename || "??"] ||= begin
-          file, dir = file_and_dir(location.filename)
-          di_builder.create_file(file, dir)
-        end
+      filename = @current_debug_location.try(&.filename) || "??"
+      debug_files_cache[filename] ||= begin
```

Comment:
This would try to look up `??` as a filename if `@current_debug_location` is nil. Does that make any sense?

Reply:
It hasn't changed since previous version, just ftr

Reply:
It has changed. Before, `??` would not be looked up if `@current_debug_location` is nil. Only if the location was present but its fielname was nil.

Reply:
I was referring to the `filename` part, sorry for being imprecise.

Reply:
`filename` is nilable, so it's really just there to satisfy a type constraint